packages/enclave-contracts/contracts/lib/ExitQueueLib.sol

```diff
@@ -0,0 +1,485 @@
+        uint64 exitDelaySeconds,
+        uint256 licenseAmount
+    ) internal {
+        queueAssetsForExit(state, operator, exitDelaySeconds, 0, licenseAmount);
+    }
+
+    /**
+     * @notice Gets the total pending amounts for an operator across all tranches
+     * @dev Returns both locked (not yet claimable) and unlocked (claimable) amounts
+     * @param state The exit queue state storage
+     * @param operator The operator to query
+     * @return ticketAmount Total pending tickets in the exit queue
+     * @return licenseAmount Total pending licenses in the exit queue
+     */
+    function getPendingAmounts(
+        ExitQueueState storage state,
+        address operator
+    ) internal view returns (uint256 ticketAmount, uint256 licenseAmount) {
+        PendingAmounts storage pending = state.pendingTotals[operator];
+        return (pending.ticketAmount, pending.licenseAmount);
+    }
+
+    /**
+     * @notice Previews the amounts that can be claimed at the current block timestamp
+     * @dev Iterates through tranches and sums up amounts where unlock timestamp has passed
+     * @param state The exit queue state storage
+     * @param operator The operator to query
+     * @return ticketAmount Total claimable tickets at current timestamp
+     * @return licenseAmount Total claimable licenses at current timestamp
+     */
+    function previewClaimableAmounts(
+        ExitQueueState storage state,
+        address operator
+    ) internal view returns (uint256 ticketAmount, uint256 licenseAmount) {
+        ExitTranche[] storage operatorQueue = state.operatorQueues[operator];
+        uint256 currentIndex = state.queueHeadIndex[operator];
+
+        for (uint256 i = currentIndex; i < operatorQueue.length; i++) {
+            ExitTranche storage tranche = operatorQueue[i];
+
+            if (block.timestamp < tranche.unlockTimestamp) {
+                break;
+            }
+
+            ticketAmount += tranche.ticketAmount;
+            licenseAmount += tranche.licenseAmount;
+        }
+    }
+
+    /**
+     * @notice Claims unlocked assets from the exit queue
+     * @dev Only processes tranches where unlock timestamp has passed. Updates pending totals
+     *      and cleans up empty tranches.
+     * @param state The exit queue state storage
+     * @param operator The operator claiming assets
+     * @param maxTicketAmount Maximum tickets to claim (actual claimed may be less if queue has fewer)
+     * @param maxLicenseAmount Maximum licenses to claim (actual claimed may be less if queue has fewer)
+     * @return ticketsClaimed Actual amount of tickets claimed
+     * @return licensesClaimed Actual amount of licenses claimed
+     */
+    function claimAssets(
+        ExitQueueState storage state,
+        address operator,
+        uint256 maxTicketAmount,
+        uint256 maxLicenseAmount
+    ) internal returns (uint256 ticketsClaimed, uint256 licensesClaimed) {
+        if (maxTicketAmount > 0) {
```

Comment:
I think we can remove these if (ticket/license amount > 0) because inside _takeAssetsFromQueue we already check if it's 0